src/Sarif/Core/ExceptionData.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// Describes a condition relevant to the tool itself, as opposed to being relevant to a file being analyzed by the tool.
+    /// </summary>
+    public partial class ExceptionData
+    {
+        public static ExceptionData Create(Exception exception)
+        {
+            return new ExceptionData
+            {
+                Type = exception.GetType().Name,
+                Message = exception.Message,
+                InnerExceptions = GetInnerExceptions(exception),
+                Stack = Stack.Create(exception.StackTrace)
+            };
+        }
+
+        private static IList<ExceptionData> GetInnerExceptions(Exception exception)
+        {
+            var innerExceptions = new List<ExceptionData>();
+
+            IReadOnlyCollection<Exception> aggregateInnerExceptions = (exception as AggregateException)?.InnerExceptions;
+            if (aggregateInnerExceptions != null)
+            {
+                foreach (Exception innerException in aggregateInnerExceptions)
```

Comment:
Consider using AggregatedException.Flatten to retrieve the inner exception closure. I think we probably want the root AggregatedException in the object model, but am not sure direct rendering of all the nested aggregated exceptions (if any exist) is super helpful.